src/compat/array/sortedIndexOf.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { describe, expect, it } from 'vitest';
+import { sortedIndexOf } from './sortedIndexOf';
+
+describe('sortedIndexOf', () => {
+  //--------------- Lodash Test Case #1 -------------------
+  it('should find first occurrence in numeric array', () => {
+    const array = [4, 4, 5, 5, 5, 6, 6];
+    expect(sortedIndexOf(array, 5)).toBe(2);
+  });
+
+  //--------------- Lodash Test Case #2 -------------------
+  it(`should return \`-1\` for an unmatched value`, () => {
+    const array = [1, 2, 3];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-expect-error
+    const empty = [];
```

Comment:
```suggestion
    const empty: unknown[] = [];
```

You can do like this!

Reply:
It looks great..! 👍